### PR TITLE
cmgd: changes to handle conversion of xpath to adapter

### PR DESCRIPTION
**Pull request by rameshabhinay**
Have implemented the changes to convert xpath string to
adapter list that can be used to dispath xpaths to specific
daemons or modules.

Signed-off-by: Abhinay Ramesh <rabhinay@vmware.com>

### CODE REVIEW

File: cmgd/cmgd_bcknd_adapter.h [hunk 1]
```diff
@@ -25,24 +25,113 @@
...
+#define CMGD_CLIENT_ID_LDPD (1 << CMGD_CLIENT_INDEX_LDPD)
+#define CMGD_CLIENT_ID_ISISD (1 << CMGD_CLIENT_INDEX_ISISD)
+#define CMGD_CLIENT_ID_FABRICD (1 << CMGD_CLIENT_INDEX_FABRICD)
+#define CMGD_CLIENT_ID_EIGRPD (1 << CMGD_CLIENT_INDEX_EIGRPD)
+#define CMGD_CLIENT_ID_BGPD (1 << CMGD_CLIENT_INDEX_BGPD)
+#define CMGD_CLIENT_ID_BABELD (1 << CMGD_CLIENT_INDEX_BABELD)
+#define CMGD_CLIENT_ID_BFDD (1 << CMGD_CLIENT_INDEX_BFDD)
+
+// XPATH regular expression map index
+typedef enum cmgd_bcknd_xpath_regexp_id_ {
```
Comment (pushpasis) [inline cmgd/cmgd_bcknd_adapter.h:76]: Why do we need regexp id?

File: cmgd/cmgd_vty.c [hunk 2]
```diff
@@ -197,10 +200,21 @@ DEFPY(show_cmgd_db,
 	return CMD_SUCCESS;
 }
 
+DEFPY(cmgd_commit, cmgd_commit_cmd, "commit-apply",
```
Comment (pushpasis) [inline cmgd/cmgd_vty.c:203]: Please create a separate commit for just the command and its processing.

File: staticd/static_vty.c [hunk 3]
```diff
@@ -1205,10 +1205,9 @@ static void static_cmgd_bcknd_client_connect(
 }
 
 static cmgd_bcknd_client_params_t cmgd_params = {
-	.name = "Staticd",
+	.name = "staticd",
```
Comment (pushpasis) [inline staticd/static_vty.c:1208]: Why change this?

**Review by pushpasis — CHANGES_REQUESTED** [review]
Please separate out a different commit for just the 'commit-apply' command.

File: cmgd/cmgd_bcknd_adapter.h (hunk 1 above)
Comment (rameshabhinay) [reply cmgd/cmgd_bcknd_adapter.h:36]: As discussed its used for readability while comparing the expression and easy to understand the derivation process.

File: staticd/static_vty.c (hunk 3 above)
Comment (rameshabhinay) [reply staticd/static_vty.c]: As discussed to keep it consistent across cmgd and modules.

**Review by pushpasis — APPROVED** [review]

**Pull request merged**